multicluster/cmd/link.go

```
@@ -457,11 +477,20 @@ func buildServiceMirrorValues(opts *linkOptions) (*multicluster.Values, error) {
 		return nil, fmt.Errorf("--log-format must be one of: plain, json")
 	}
 
+	if opts.disableGateway {
+		if opts.selector == fmt.Sprintf("%s=%s", k8s.DefaultExportedServiceSelector, "true") {
+			opts.selector = ""
+		} else if opts.selector != "" {
+			return nil, fmt.Errorf("--selector and --no-gateway are mutually exclusive, received selector: %s", opts.selector)
```

we should also return an error if any of the `--gateway-*` flags are specified.